[24.1] Handle special charater in raw SQL

jdavcs opened the pull request:
Fix #19880

This statement is executed by SQLAlchemy and also via psycopg in the `pgcleanup.py` script. For psycopg, the `%` character should be replaced with `%%` (ref https://www.psycopg.org/docs/usage.html#passing-parameters-to-sql-queries) SQLAlchemy already takes care of it automatically.

Executing the statement with the double % via SQLAlchemy (which will double them) won't cause any problems.


## How to test the changes?
(Select all options that apply)
- [ ] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [ ] This is a refactoring of components with existing test coverage.
- [ ] Instructions for manual testing are as follows:
  1. [add testing steps and prerequisites here if you didn't write automated tests covering all your changes]

## License
- [x] I agree to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT).

bernt-matthias reviewed (approved):
Successfully tested on my instance. Thanks a lot.

nsoranzo committed:
Add type annotations to ``calculate_user_disk_usage_statements()``

and related code.

jdavcs committed:
Handle "%" character in raw SQL

This statement is prepared for SQLAlchemy but executed via psycopg2 in
the `pgcleanup.py` script. For psycopg2, the "%" character should be replaced
with %% (see docs
https://www.psycopg.org/docs/usage.html#passing-parameters-to-sql-queries).
SQLAlchemy already takes care of it automatically.
